src/lib/lore.sh

```diff
@@ -276,72 +312,146 @@ function reset_list_of_mailinglist_patches()
-  local count=0
-  local index=0
-  local title
-  local url_filter='?q=d:2.day.ago..+AND+NOT+s:Re&x=A'
-  local default_url="${LORE_URL}${target_mailing_list}/${url_filter}"
-  local shared_dir_for_parallelism
-  local list_patches_file_name="list-patches-${target_mailing_list}.xml"
-
-  download "$default_url" "$list_patches_file_name" "$CACHE_LORE_DIR" "$flag" || return "$?"
-
-  raw_list_path="${CACHE_LORE_DIR}/${list_patches_file_name}"
 
   xpath_query="${NAME_EXP}|${EMAIL_EXP}|${TITLE_EXP}|${LINK_EXP}"
+  printf '%s' "$raw_xml" | xpath -q -e "$xpath_query"
+}
 
-  pre_processed=$(< "$raw_list_path")
-  pre_processed=$(printf '%s' "$pre_processed" | xpath -q -e "$xpath_query")
+# This function converts a list of patches into a list of patchsets stored
+# in the `list_of_mailinglist_patches` array. A patchset differs from a
+# single patch, because the first includes all patches in a series of patches
+# which can have different version. For each multipart patchset, the first patch
+# is either a cover letter or a actual patch and is the representative of the
+# patchset. This patch metadata is the one stored in `list_of_mailinglist_patches`.
+#
+# @pre_processed_patches: String containing a list of pre-processed patches
+# TODO:
+# - The function `is_introduction_patch` basically filters which patch is a
+#   representative of the patchset by the message-ID. Some valid representatives
+#   are wrongly filtered out, because of what the function considers a message-ID
+#   from a representative.
+# - The function `extract_metadata_from_patch_title` called by `thread_for_process_patch`
+#   counts the cover letter as a patch which results in patchsets with cover letters
+#   having one more patch than in reality.
+function process_patchsets()
+{
+  local pre_processed_patches="$1"
+  local shared_dir_for_parallelism
+  local starting_index
+  local line
+  local processed_patchset
+  local patch_url
+  local patch_title
+  local count
+
+  shared_dir_for_parallelism=$(mktemp --directory)
+  starting_index="$PATCHSETS_PROCESSED"
+  count=0
 
-  # Converting to:
-  #  Author, email, version, total patches, patch title, link
-  shared_dir_for_parallelism=$(mktemp -d)
   while IFS= read -r line; do
-    if [[ "$line" =~ href ]]; then
-      message_id_link=$(str_get_value_under_double_quotes "$line")
+    if [[ "$line" =~ href && "$PATCHSETS_PROCESSED" -lt "$NUMBER_OF_PATCHSETS_PER_FETCH" ]]; then
+      patch_url=$(str_get_value_under_double_quotes "$line")
 
-      if is_introduction_patch "$message_id_link"; then
+      if is_introduction_patch "$patch_url"; then
         # Process each patch in parallel
-        thread_for_process_patch "$index" "$shared_dir_for_parallelism" \
-          "$processed_line" "$message_id_link" "$title" &
-        ((index++))
+        thread_for_process_patch "$PATCHSETS_PROCESSED" "$shared_dir_for_parallelism" \
+          "$processed_patchset" "$patch_url" "$patch_title" &
+        ((PATCHSETS_PROCESSED++))
```

Comment:
In a later commitl, this becomes more evident, but my intetion was to use this global variable to hold the "state" across many calls of `fetch_latest_patchsets_from` (and its subroutines). We could shift the responsability to the caller (`PATCHSETS_PROCESSED` becomes an argument for `fetch_latest_patchsets_from` and its subroutines), but I thought this broke the MVC pattern that is loosely implemented and because I think the caller would have to use a global variable also.